tests/src/Common/CoreFX/TestFileSetup/Program.cs

```diff
@@ -0,0 +1,141 @@
+
+                    if (!File.Exists(dotnetPath))
+                        throw new ArgumentException("Invalid testhost path. Please supply a test host location to run tests.");
+
+                    exitCode = RunTests();
+
+                }
+            }
+            catch (AggregateException e)
+            {
+                e.Handle(innerExc =>
+                {
+
+                    if (innerExc is HttpRequestException)
+                    {
+                        exitCode = ExitCode.HttpError;
+                        Console.WriteLine("Error downloading tests from: " + testUrl);
+                        Console.WriteLine(innerExc.Message);
+                        return true;
+                    }
+                    else if (innerExc is IOException)
+                    {
+                        exitCode = ExitCode.IOError;
+                        Console.WriteLine(innerExc.Message);
+                        return true;
+                    }
+                    else if (innerExc is JSchemaValidationException || innerExc is JsonSerializationException)
+                    {
+                        exitCode = ExitCode.JsonSchemaValidationError;
+                        Console.WriteLine("Error validating test list: ");
+                        Console.WriteLine(innerExc.Message);
+                        return true;
+                    }
+                    else
+                    {
+                        exitCode = ExitCode.UnknownError;
+                    }
+                    return false;
+                });
+            }
+
+            Environment.Exit((int)exitCode);
+        }
+
+
+        private static ArgumentSyntax ParseCommandLine(string[] args)
+        {
+            ArgumentSyntax argSyntax = ArgumentSyntax.Parse(args, syntax =>
+            {
+                syntax.DefineOption("out|outDir|outputDirectory", ref outputDir, "Directory where tests are downloaded");
+                syntax.DefineOption("testUrl", ref testUrl, "URL, pointing to the list of tests");
+                syntax.DefineOption("testListJsonPath", ref testListPath, "JSON-formatted list of test assembly names to download");
+                syntax.DefineOption("clean|cleanOutputDir", ref cleanTestBuild, "Clean test assembly output directory");
+                syntax.DefineOption("run|runTests", ref runTests, "Run Tests after setup");
+                syntax.DefineOption("dotnet|dotnetPath", ref dotnetPath, "Path to dotnet executable used to run tests.");
+                syntax.DefineOption("executable|executableName", ref executableName, "Name of the test executable to start");
+                syntax.DefineOption("log|logPath|logRootOutputPath", ref logRootOutputPath, "Run Tests after setup");
+                syntax.DefineOption("maxProcessCount|numberOfParallelTests|maximumDegreeOfParalellization", ref maximumDegreeOfParalellization, "Maximum number of concurrently executing processes");
```

Comment:
What is the reason for having multiple names for this option? Just curious...

Reply:
No particular reason. Makes it a bit easier to use (don't have to remember the exact phrasing)